index.js

```diff
@@ -3,37 +3,26 @@ const { isTriggerableBranch } = require('./lib/triggerable-branch')
 const { findReleases, generateReleaseInfo } = require('./lib/releases')
 const { findCommitsWithAssociatedPullRequests } = require('./lib/commits')
 const { validateReplacers } = require('./lib/template')
-const {
-  validateSortDirection,
-  sortPullRequests,
-  SORT_DIRECTIONS
-} = require('./lib/sort-pull-requests')
+const { sortPullRequests } = require('./lib/sort-pull-requests')
+const { schema } = require('./lib/schema')
 const log = require('./lib/log')
 
 const configName = 'release-drafter.yml'
 
 module.exports = app => {
   app.on('push', async context => {
-    const defaults = {
-      branches: context.payload.repository.default_branch,
-      'change-template': `* $TITLE (#$NUMBER) @$AUTHOR`,
-      'no-changes-template': `* No changes`,
-      'version-template': `$MAJOR.$MINOR.$PATCH`,
-      categories: [],
-      'exclude-labels': [],
-      replacers: [],
-      'sort-direction': SORT_DIRECTIONS.descending
+    let config = {}
+    try {
+      const repoConfig = (await getConfig(context, configName)) || {}
+      const { error, value } = schema(context).validate(repoConfig)
+      if (error) {
+        throw error
+      }
+      config = value
+    } catch (error) {
+      log({ app, context, error, message: 'Invalid config file' })
+      return
     }

```

Comment:
I'd like to extract this to a separate module, just like https://github.com/release-drafter/release-drafter/pull/338. Though I can also do this after this is merged :)